src/index.ts

```diff
@@ -80,6 +82,8 @@ export interface RequestCallback<T = any> {
   (err: Error | null, response: Response, body?: T): void;
 }
 
+// tslint:disable-next-line variable-name
+const HttpProxyAgent = require('http-proxy-agent');
```

Comment:
For both of these ...  I think we should delay the require.  For 90%+ of cases here, I'd guess folks *aren't* using proxies.  For those cases, I would prefer to lazy load the module required.